File: crates/harn-hostlib/tests/fixtures/ast/scala/source.scala
trait Speaker {
  def speak(): String
}

case class Greeter(name: String) extends Speaker {
  def speak(): String = s"hello $name"
}

object Defaults {
  val World: Greeter = Greeter("world")
}

def shout(s: String): String = s.toUpperCase

File: crates/harn-hostlib/tests/fixtures/ast/kotlin/source.kt
interface Speaker {
    fun speak(): String
}

class Greeter(val name: String) : Speaker {
    override fun speak(): String {
        return "hello $name"
    }

    fun shout(message: String): String {
        return message.uppercase()
    }
}

object Defaults {
    fun world(): Greeter = Greeter("world")
}

fun globalShout(s: String): String = s.uppercase()
